lib/Doctrine/Common/DataFixtures/ReferenceRepository.php

```diff
@@ -136,10 +167,12 @@ public function setReferenceIdentity($name, $identity)
      */
     public function addReference($name, $object)
     {
-        if (isset($this->references[$name])) {
+        $class = get_class($object);
+        if (isset($this->referencesByClass[$class][$name])) {
```

Comment:
This change allows to use `addReference` with a different class name to override a previous name, which is then a BC break (for code that get reference by name only and so is relying on name uniqueness globally and not per class), as you silently change the behavior of that method (which could now be documented as `$name must not be set yet for the object class` instead of `$name must not be set yet`).

Reply:
Indeed, I added the check back with a comment to remove it in next major.